src/components/Property/Types/VectorProperty/ViewOptions/ColorView.tsx

```diff
@@ -20,38 +39,71 @@ export function ColorView({
   additionalData,
   isInt
 }: Props) {
-  if ((value.length !== 3 && value.length !== 4) || isInt) {
-    throw Error('Invalid use of Color view option!');
-  }
+  const [isError, setIsError] = useState(false);
+
+  const isOutsideRange = value.some((v) => v < 0 || v > 1);
+  const currentColor: RGBA = openspaceColorToRgba(value);
   const hasAlpha = value.length === 4;
+  const valueDisplayString = value
+    .map((v) => parseFloat(v.toFixed(3)).toString())
+    .join(', ');
+
+  validateInput(value, additionalData, isInt);
+
+  function isValidColorString(colorString: string): boolean {
+    const newValues = colorString.split(',').map((v) => parseFloat(v));
+    const result =
+      newValues.length === value.length &&
+      newValues.every((v) => !isNaN(v) && isNumber(v));
+
+    return result;
+  }
+
+  function onColorPickerChange(color: RGBA) {
+    const newValue = toOpenspaceColor(color, hasAlpha);
+    setPropertyValue(newValue);
+
+    setIsError(!isValidColorString(newValue.join(', ')));
```

Comment:
How about moving the `.join(', ')` part into the `isvalidColor` function? 

Or, better, remove it completely? Seems like the first thing we do in the function is to split based on the `,` again :)

Reply:
The function is used in four places, where two of them are strings like this so I thinks it's better to keep it the way it is